Update qos.rst after first UI release (#24)
NethServer/nethsecurity-docs: qos.rst

Quality of Service (QoS)
=========================

NethSecurity QoS provides Active Queue Management (AQM) and Flow Queuing (FQ) capabilities to help ensure that your network resources are used efficiently and fairly.

Operating principles
====================

NethSecurity QoS is designed to make the best use of available bandwidth, without imposing strict limits or shaping traffic by default.
It operates on the following principles:

- *Bandwidth usage*: QoS strives to utilize the available bandwidth to its fullest potential. By default, it does not impose strict bandwidth limitations on your network.
  Instead, it dynamically adjusts to network conditions, ensuring that unused bandwidth is efficiently utilized.

- *Flow management*: QoS actively manages network flows to prevent any single client or application from monopolizing the available bandwidth.
  This ensures fair and equitable access to network resources for all users.

Configuration
=============

Bandwidth management is handled dynamically and automatically by the system. Configuration is straightforward, involving specifying upload and download bandwidth values for each interface under QoS. 

> While QoS can be set up on any interface, it typically functions optimally on WAN-type interfaces, aligning upload and download data with connectivity parameters.

Reviewer: ```suggestion
While QoS can be set up on any interface, it typically functions optimally on WAN-type interfaces, setting upload and download speeds to the internet connection data rates.
```

To ensure resilience against service fluctuations, it is advisable to maintain a safety margin by configuring these parameters 10% lower than the measured values

Advanced Configuration

> ====================

Reviewer 2: ```suggestion
====================

QoS relies on an eBPF (Extended Berkeley Packet Filter) based classifier to set Differentiated Services Code Point (DSCP) fields in packets. This classification helps prioritize and manage network traffic efficiently.
To maximize efficiency, QoS operates in the kernel space using eBPF technology. This ensures minimal overhead and minimal impact on system performance.
In addition to IP and port-based rules, QoS allows you to define traffic rules based on DNS names, providing granular control over how traffic is classified and treated.
```

While Qosify works effectively without extensive configuration, it can be further optimized by setting bandwidth limits and rules.
Fine-tuning QoS parameters can result in even better network performance.

The modifications to the standard behavior however can be useful in very limited scenarios, for which, currently, it is only possible to act directly through the character console.

Priority classes
----------------

> The QoS uses four classes of priority, each class can use a maximum percentage of bandwidth defined by its threshold.

Reviewer: ```suggestion
QoS uses four classes of priority, each class can use a maximum percentage of bandwidth defined by its threshold.
```
